miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultStorageLocationService.java

```diff
@@ -216,35 +220,29 @@ public void authorizeDeletion(StorageLocation object) throws IOException {
     authorizationManager.throwIfNonAdminOrMatchingOwner(object.getCreator());
   }
 
+  private static final Set<LocationUnit> DELETABLE_FROM = Collections
+      .unmodifiableSet(Sets.newHashSet(LocationUnit.ROOM, LocationUnit.FREEZER, LocationUnit.SHELF));
+
   @Override
   public ValidationResult validateDeletion(StorageLocation object) throws IOException {
     ValidationResult result = new ValidationResult();
-    switch (object.getLocationUnit()) {
-    case ROOM:
+    if (object.getLocationUnit() == LocationUnit.ROOM) {
       if (!object.getChildLocations().isEmpty()) {
-        result.addError(
-            new ValidationError(String.format("Room %s contains %d freezers", object.getAlias(), object.getChildLocations().size())));
+        result.addError(new ValidationError(String.format("Room %s contains %d %s", object.getAlias(), object.getChildLocations().size(),
+            Pluralizer.freezers(object.getChildLocations().size()))));
       }
-      break;
-    case FREEZER:
-      int boxes = countBoxes(object);
-      if (boxes > 0) {
-        result.addError(new ValidationError(String.format("Freezer %s contains %d %s", object.getAlias(), boxes, Pluralizer.boxes(boxes))));
+    } else {
+      if (object.getParentLocation() != null && !DELETABLE_FROM.contains(object.getParentLocation().getLocationUnit())) {
+        result.addError(new ValidationError("This storage unit cannot be deleted directly"));
```

Comment:
If the user actually sees this message, they're doing something they shouldn't be, since the delete button is hidden for anything that can't be deleted directly (rack/stack positions). This is just here to make sure something weird doesn't happen